logic/LogicManager.java

```diff
@@ -103,6 +107,23 @@ private FunctionalIterator<Rule> rulesWithNegations() {
         return rules().filter(rule -> !rule.when().negations().isEmpty());
     }
 
+    public Map<Rule, Set<Unifier>> applicableRules(Concludable concludable) {
+        Map<Rule, Set<Unifier>> unifiers = logicCache.unifiers().getIfPresent(concludable);
+        if (unifiers == null) {
+            unifiers = concludable.computeApplicableRules(conceptMgr, this);
+        }
+        return unifiers;
+    }
+
+    public void indexConcludables(ResolvableConjunction conjunction) {
+        link(Iterators.single(conjunction),
+                iterate(conjunction.negations()).flatMap(negation -> iterate(negation.disjunction().conjunctions())))
```

Comment:
this can be symmetric to `applicableRules(concludable)` to be something like `indexApplicableRules(concludable)` - the iteration can be moved externally, into `ResolvableConjunction`